R/messages.R

```diff
@@ -0,0 +1,62 @@
+errorWrongType <- function(object, optionalMessage = NULL) {
+  # Name of the calling function
+  callingFunctions <- sys.calls()
+  callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
+
+  # Name of the variable in the calling function
+  objectName <- deparse(substitute(object))
+
+  message <- paste0(
+    callingFunction, ": argument '", objectName,
+    "' is of wrong type! Type in '?", callingFunction,
+    "' for information on required types", optionalMessage
+  )
+
+  return(message)
+}
+
+#' Error message when single output is expected but multiple entities produced.
+#'
+#' @param path A vector of strings relative to the \code{container}
+#' @param container A Container or Simulation used to find the entities
+#' @param optionalMessage A string that will be appended to the end of the message
+#'
+#' @return A string representation of the error message
+errorGetEntityMultipleOutputs <- function(path, container, optionalMessage = NULL) {
+  # Name of the calling function
+  callingFunction <- sys.call(-2)[[1]]
```

Comment:
Can you explain where is the -2 etc. is coming from. This will be very useful to understand the code in a few weeks from now

Reply:
It actually should read 

```
  callingFunctions <- sys.calls()
  callingFunction <- sys.call(-length(callingFunctions) + 1)[[1]]
```

The negative number makes to traverse the call stack backwards. +1 was more of a trial-and-error to figure out how to dynamically get the right name.

`callingFunction <- sys.call(-2)[[1]]`
was from stackoverflow and I do not understand completely why -2 so far...